controller/contour/namespace.go

```diff
@@ -58,6 +64,12 @@ func (r *Reconciler) ensureNamespaceRemoved(ctx context.Context, name string) er
 	switch {
 	case err == nil:
 		if err := r.Client.Delete(ctx, ns); err != nil {
+			// protect against a race condition between contour being updated
+			// while the namespace is being deleted.
+			if errors.IsGone(err) {
```

Comment:
```suggestion
			if errors.IsNotFound(err) {
```
I've always used `IsNotFound` for this kind of check.  Grepping the Kubernetes codebase turns up only a couple uses of `IsGone`; it looks like it is needed only for an esoteric and deprecated API.